File: frontEnd/src/types/dto/stepDTO/ScriptStepDTO.ts
import {StepBaseDTO} from "@/types/dto/base/StepBaseDTO";

/**
 * 脚本步骤DTO
 *
 * 在场景变量池上下文中执行 JS（GraalJS 沙箱）：
 * - context.getVariable / setVariable 读写场景变量
 * - console.log 输出日志
 * - context.assertCondition 自定义断言
 */
export class ScriptStepDTO extends StepBaseDTO {
    stepType: string = 'SCRIPT';

    /**
     * JS 脚本内容
     */
    scriptContent: string = '';
}

File: backEnd/platform-automation/src/main/java/com/mokatest/platform/demos/domain/ui/dto/step/ScriptStepDTO.java
package com.mokatest.platform.demos.domain.ui.dto.step;

import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 脚本步骤（在场景变量池上下文中执行 JS）
 */
@Data
@EqualsAndHashCode(callSuper = true)
public class ScriptStepDTO extends StepBaseDTO {

    private String stepType;

    /**
     * JS 脚本内容（可读写 context 变量、console.log、context.assertCondition）
     */
    private String scriptContent;
}
